app/controllers/tasks_controller.rb

```diff
@@ -1,18 +1,25 @@
 class TasksController < ApplicationController
 
   def create
-    @mentor = current_user.partner
-    @project = Project.find_by(mentee_id: current_user.id)
+    @project = current_user.project
+    current_user.mentee? ? (@mentor = @project.mentor) : (@mentee = @project.mentee)
```

Comment:
Check if it possible to use current_user.partner rather than @mentor/@mentee declaration :)